tests/test_requests.py

```diff
@@ -17,7 +17,7 @@ def async_mock():
     """Workaround since python 3.7 doesn't ship with asyncmock."""
 
     async def async_test(req):
-        return HttpResponse(req.url, body=b"")
+        return HttpResponse(str(req.url), body=b"")
```

Comment:
I don't think we should be casting this to `str` as the tests that rely on this mock are expecting `ResponseUrl`. We'd need to update the downstream tests instead to validate their expectations.

Reply:
I must admit I failed to make this test working, and applied this hack instead :) Mocks are not my friends.

Reply:
I was thinking it's harmless, because tests should be getting ResponseUrl, as HttpResponse should be converting str to ResponseUrl; without having mocks in mind the change doesn't really change the behavior. But I'm likely missing some interactions with mocks which are in effect here.